tests/test_sync.py

```diff
@@ -158,12 +158,14 @@ def test_ApiSync_on_sync_failure(mocker, session_maker, homedir):
     """
     api_sync = ApiSync(mocker.MagicMock(), session_maker, mocker.MagicMock(), homedir)
     sync_failure = mocker.patch.object(api_sync, "sync_failure")
+    singleShot_fn = mocker.patch("securedrop_client.sync.QTimer.singleShot")
 
     error = Exception()
 
     api_sync.on_sync_failure(error)
 
     sync_failure.emit.assert_called_once_with(error)
+    singleShot_fn.assert_called_once_with(15000, api_sync.api_sync_bg_task.sync)
```

Comment:
In relation to my comment above: testing time-related things is difficult, so this seems like an OK compromise to me even though it tests the implementation more than the outcome.

Reply:
yeah, i went back and forth once on this, but figured we'd want to make sure this "business rule" of 15 seconds is confirmed.

Reply:
I agree it's worth having some guarantees at least, and this code is unlikely to become high-traffic (as in change a lot), so there is less downside to testing its implementation directly. :+1: 

My last attempt at testing time-dependent changes resulted in https://github.com/gonzalo-bulnes/stime :wink: